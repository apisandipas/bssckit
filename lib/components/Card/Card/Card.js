"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _theme = _interopRequireDefault(require("../../../theme"));

var _base = require("../../Utilities/base");

var _themeFunctions = require("../../../utils/themeFunctions");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n  height: 100%;\n  position: relative;\n  display: flex;\n  flex-direction: column;\n  min-width: 0;\n  word-wrap: break-word;\n  background-color: ", ";\n  background-clip: border-box;\n  border: ", " ", ";\n  ", "\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n    border-radius: ", ";\n  "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n      border-radius: ", ";\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var borderRadius = function borderRadius(props) {
  if (props.noRadius) {
    return (0, _styledComponents.css)(_templateObject(), (0, _themeFunctions.getBorderRadius)(props, 'card', 'noRadius'));
  }

  return (0, _styledComponents.css)(_templateObject2(), (0, _themeFunctions.getBorderRadius)(props, 'card', 'default'));
};

var Card = (0, _styledComponents["default"])(_base.Div)(_templateObject3(), function (props) {
  return (0, _themeFunctions.getColor)(props, 'card', 'backgroundColor');
}, function (props) {
  return (0, _themeFunctions.getBorder)(props, 'card', 'default');
}, function (props) {
  return (0, _themeFunctions.getColor)(props, 'card', 'borderColor');
}, function (props) {
  return borderRadius(props);
});
Card.defaultProps = {
  theme: _theme["default"]
};
var _default = Card;
exports["default"] = _default;