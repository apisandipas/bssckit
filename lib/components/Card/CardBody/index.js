"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "CardBody", {
  enumerable: true,
  get: function get() {
    return _CardBody["default"];
  }
});

var _CardBody = _interopRequireDefault(require("./CardBody"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }