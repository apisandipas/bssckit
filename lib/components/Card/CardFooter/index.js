"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "CardFooter", {
  enumerable: true,
  get: function get() {
    return _CardFooter["default"];
  }
});

var _CardFooter = _interopRequireDefault(require("./CardFooter"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }