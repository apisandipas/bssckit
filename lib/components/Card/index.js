"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Card", {
  enumerable: true,
  get: function get() {
    return _Card.Card;
  }
});
Object.defineProperty(exports, "CardBody", {
  enumerable: true,
  get: function get() {
    return _CardBody.CardBody;
  }
});
Object.defineProperty(exports, "CardFooter", {
  enumerable: true,
  get: function get() {
    return _CardFooter.CardFooter;
  }
});
Object.defineProperty(exports, "CardHeader", {
  enumerable: true,
  get: function get() {
    return _CardHeader.CardHeader;
  }
});
Object.defineProperty(exports, "CardImageHeader", {
  enumerable: true,
  get: function get() {
    return _CardImageHeader.CardImageHeader;
  }
});
Object.defineProperty(exports, "CardText", {
  enumerable: true,
  get: function get() {
    return _CardText.CardText;
  }
});
Object.defineProperty(exports, "CardTitle", {
  enumerable: true,
  get: function get() {
    return _CardTitle.CardTitle;
  }
});

var _Card = require("./Card");

var _CardBody = require("./CardBody");

var _CardFooter = require("./CardFooter");

var _CardHeader = require("./CardHeader");

var _CardImageHeader = require("./CardImageHeader");

var _CardText = require("./CardText");

var _CardTitle = require("./CardTitle");