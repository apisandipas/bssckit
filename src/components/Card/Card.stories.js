import React from 'react';
import { Button } from '../Button';
import { Card, CardBody, CardImageHeader, CardTitle, CardText } from './';

export default {
  title: 'Card',
};

export const simple = () => (
  <Card style={{ width: '18rem' }}>
    <CardImageHeader src='http://via.placeholder.com/350x150' />
    <CardBody>
      <CardTitle h3>Card title</CardTitle>
      <CardText>
        Some quick example text to build on the card title and make up the bulk of the card&#39;s content.
      </CardText>
      <Button primary>Go somewhere</Button>
    </CardBody>
  </Card>
);

export const withPlainBody = () => (
  <Card>
    <CardBody>This is some text within a card body.</CardBody>
  </Card>
);

export const withMultipleTitles = () => (
  <Card style={{ width: '18rem' }}>
    <CardBody>
      <CardTitle h3>Card title</CardTitle>
      <CardTitle h4 muted subtitle>
        Card subtitle
      </CardTitle>
      <CardText>
        Some quick example text to build on the card title and make up the bulk of the card&#39;s content.
      </CardText>
    </CardBody>
  </Card>
);

export const withLinks = () => (
  <Card style={{ width: '18rem' }}>
    <CardBody>
      <CardTitle h3>Card title</CardTitle>
      <CardText>
        Some quick example text to build on the card title and make up the bulk of the card&#39;s content.
      </CardText>
      <a href=''>Card Link</a>
      <a href=''>Another link</a>
    </CardBody>
  </Card>
);
