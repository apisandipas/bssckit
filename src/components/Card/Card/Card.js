import styled, { css } from 'styled-components';
import { theme } from 'styled-tools';
import defaultTheme from '../../../theme';
import { Div } from '../../Utilities/base';

const getBorderRadius = props => {
  if (props.noRadius) {
    return css`
      border-radius: ${theme('card.borderRadius.noRadius')};
    `;
  }

  return css`
    border-radius: ${theme('card.borderRadius.default')};
  `;
};

const Card = styled(Div)`
  height: 100%;
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: ${theme('card.colors.default.backgroundColor')};
  background-clip: border-box;
  border: ${theme('card.border.default')} ${theme('card.colors.default.borderColor')};
  ${props => getBorderRadius(props)}
`;

Card.defaultProps = {
  theme: defaultTheme,
};

export default Card;
