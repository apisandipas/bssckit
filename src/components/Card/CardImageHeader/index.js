export { default as CardImageHeader } from './CardImageHeader';
