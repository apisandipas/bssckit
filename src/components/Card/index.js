export { Card } from './Card';
export { CardBody } from './CardBody';
export { CardFooter } from './CardFooter';
export { CardHeader } from './CardHeader';
export { CardImageHeader } from './CardImageHeader';
export { CardText } from './CardText';
export { CardTitle } from './CardTitle';
