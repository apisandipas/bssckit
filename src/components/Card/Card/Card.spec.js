import React from 'react';
import { axe } from 'jest-axe';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import Card from './Card';

describe('Card Component', () => {
  it('matches Card', async () => {
    const { container } = render(<Card />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches Card noRadius', () => {
    const { container } = render(<Card noRadius />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('border-radius', '0');
  });
});
