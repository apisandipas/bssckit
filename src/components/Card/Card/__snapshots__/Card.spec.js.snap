// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches Card 1`] = `
.c0 {
  height: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0,0,0,0.2);
  border-radius: 0.25rem;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
/>
`;

exports[`Card Component matches Card noRadius 1`] = `
.c0 {
  height: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0,0,0,0.2);
  border-radius: 0;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
/>
`;
